File: mobile/lib/models/promotion.dart
import 'product.dart';

class Promotion {
  final int promotionId;
  final String title;
  final String? description;
  final String? imagePath;
  final String? imageUrl;
  final bool isActive;
  final int? productId;
  final Product? product;
  final DateTime? startDate;
  final DateTime? endDate;

  Promotion({
    required this.promotionId,
    required this.title,
    this.description,
    this.imagePath,
    this.imageUrl,
    this.isActive = true,
    this.productId,
    this.product,
    this.startDate,
    this.endDate,
  });

  factory Promotion.fromJson(Map<String, dynamic> json) {
    String? imageUrl = json['image_url'] as String?;
    if (imageUrl != null && imageUrl.contains('localhost')) {
      imageUrl = imageUrl.replaceFirst('localhost', '127.0.0.1');
    }

    return Promotion(
      promotionId: json['promotion_id'] as int,
      title: json['title'] as String,
      description: json['description'] as String?,
      imagePath: json['image_path'] as String?,
      imageUrl: imageUrl,
      isActive: json['is_active'] == true || json['is_active'] == 1,
      productId: json['product_id'] as int?,
      product: json['product'] != null
          ? Product.fromJson(json['product'] as Map<String, dynamic>)
          : null,
      startDate: json['start_date'] != null
          ? DateTime.parse(json['start_date'] as String)
          : null,
      endDate: json['end_date'] != null
          ? DateTime.parse(json['end_date'] as String)
          : null,
    );
  }
}

File: deploy_temp/backend/app/Models/Promotion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Promotion extends Model
{
    protected $primaryKey = 'promotion_id';

    protected $fillable = [
        'title',
        'description',
        'image_path',
        'is_active',
        'start_date',
        'end_date',
        'product_id',
    ];

    protected $casts = [
        'is_active' => 'boolean',
        'start_date' => 'datetime',
        'end_date' => 'datetime',
    ];

    protected $appends = ['image_url'];

    public function getImageUrlAttribute()
    {
        return $this->image_path ? asset('storage/' . $this->image_path) : null;
    }

    public function product()
    {
        return $this->belongsTo(Product::class, 'product_id', 'product_id');
    }
}
